test/unit/serialization/structure.spec.js

```diff
@@ -107,5 +107,35 @@ describe('serialization', () => {
         });
       });
     });
+    context('when static toJSON is provided', () => {
+      var Person;
+
+      beforeEach(() => {
+        Person = attributes({
+          name: String,
+          age: Number
+        })(class Person {
+          static toJSON(json) {
```

Comment:
I'm not entirely convinced the name of this method should be `toJSON` :thinking:, gonna comment it in the original issue so we discuss is there.